### ISSUE TITLE
Screen review does not work properly in windows 10 and newest updates of several applications

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#7760]
Steps to reproduce:
1. Press NVDA + numpad 1 or 7 (desktop layout) or NVDA + page up and page down to change between review modes
2. Try to navigate the screen in screen review mode

Expected behavior:
In screen review , NVDA should allow me to navigate the whole active window and should read the text which is visible on the screen.

Actual behavior:
In all applications in Winbdows 10 NVDA says only "blank blank". No text is being recognized. However, in Windows 8.1 NVDA's screen review works properly in MS Office 2010 or Innova phone. In all browsers, Screen review does not work at all.

System configuration:
NVDA version:
all

NVDA Installed or portable:
both

Other information:

Windows version:
Windows 10 fall creators, Windows 8.1

Name and version of other software in use when reproducing the issue:
several browsers, MS Office 2016, other applications

Other questions:

Does the issue still occur after restarting your PC?
yes

Have you tried any other versions of NVDA?
yes until 2013.2

**Comment by ruifontes** [comment]
Please, before opening a ticket, please read the manual...

In this case, please read section 5.6.2 about document review.

Regarding screen reader, I think the fault is not from NVDA, but from the 
way the new apps are made. Normally, with object review is possible to read 
those apps...

Rui Fontes



-----Mensagem Original----- 
De: Adriani90
Data: 14 de novembro de 2017 23:01
Para: nvaccess/nvda
Cc: Subscribed
Assunto: [nvaccess/nvda] Screen review and document review do not work 
properly in windows 10 and newest updates of several applications (#7760)



Steps to reproduce:

Press NVDA + numpad 1 or 7 (desktop layout) or NVDA + page up and page down 
to change between review modes
Try to navigate the screen in screen review mode
Try it in document review

Expected behavior:
In screen review , NVDA should allow me to navigate the whole active window. 
In document review just the section where the system caret could be moved by 
pressing tab or arrow keys in the usual navigation mode.

Actual behavior:
These two review modes are very inconsistent in several applications. While 
screen review does not work at all anymore, at least not comparable with 
earlier versions of windows or other applications, the document review works 
just in browsers but not in other applications.

System configuration:
NVDA version:
all

NVDA Installed or portable:
both

Other information:

Windows version:
Windows 10 fall creators

Name and version of other software in use when reproducing the issue:
several browsers, MS Office, other applications

Other questions:

Does the issue still occur after restarting your PC?
yes

Have you tried any other versions of NVDA?
yes until 2013.2

**Comment by Adriani90** [comment]
Hey Rui, thanks for your comment. I have red the manual before. But it was a mistake which i have done for document review. It works properly. But screen review does not work. I have just edited my ticket accordingly.

You are right, in most cases you can navigate by object navigation, but in some cases screen review is much faster since it gives you an overview of more objects in a very short time. It would be a shame if this very good method of navigation is being abandoned just because applications come with new technologies. Let's discuss about how to improve it.

**Comment by ruifontes** [comment]
Hi Adriani!

You wrote:
You are right, in most cases you can navigate by object navigation, but in 
some cases screen review is much faster since it gives you an overview of 
more objects in a very short time. It would be a shame if this very good 
method of navigation is being abandoned just because applications come with 
new technologies.

I don't believe that screen review is being abandoned...
It seems more a impossibility caused by the new ways text is displayed on 
screen...
Jaws cursor have the same problem...


You wrote:
Let's discuss about how to improve it.

I do not have enough knowledge to discuss that!
Maybe in six months...

Rui

**Comment by josephsl** [comment]
Hi, @MichaelDCurran, any expert opinions are appreciated. Thanks.

**Comment by feerrenrut** [comment]
@Adriani90 Could you please edit the description on this issue, please update the steps to reproduce so they are specific to an application that demonstrates the issue you encounter. Also specify the speech you actually got from NVDA (or specify 'no speech') as well as the actual speech you expect? If you wish to describe several applications, you may provide multiple sections of "steps to reproduce" "actual behaviour" "expected behaviour". One section for each application.

This is helpful for us to understand the kind of problem you are seeing, if NVDA is reporting the wrong object, it is a very different problem from NVDA reporting no object. By having the information specific to a single application, we can confirm that we get the same behaviour, and try to investigate it further. It allows us to collect more information about that application, and why it's not working as expected on your behalf.

**Comment by Adriani90** [comment]
Well the approach is the same in every application. I expect, when in screen review, NVDA to let me explore the application window  from from very up to very down. Objects do not have to be recognized as such, but all the text. For example in MS Outlook I have several objects on the same line and in Screenreview on Windows 8.1 and Outlook 2010 NVDA says for example following:
Insert Address book Name File Element Signature Zoom

So I can go to the object signatur for example and activate it. But in later versions on Windows 10 NVDA does not report any text at all. I didn't test Office 2016 on Windows 8.1 to compare it but I guess it will be the same issue. I will try to test for instance Office 2010 on Windows 10 to provide a more exact result.

**Comment by Adriani90** [comment]
I have edited the description. Hope it helps.

**Comment by feerrenrut** [comment]
I have not been able to reproduce the problem. I have tested screen review and object review with NVDA version `next-14617,ef52ddc4` and in MS Office 2013 Outlook on Windows 10 1703. I dont get "blank" reported at all when in either object review or screen review.

I understand that you are getting problematic behaviour in several applications, however, it confuses the issue if you talk about them all at once. Try to make a bug report as specific as possible. Please be as clear as possible as to the differences in behaviour that you observe.

**Comment by Adriani90** [comment]
As I said, it occurs on last updates of several applications. Firefox 56 and later, Office 2016 and so on. I have really shown that it works in Office 2010 and I expected it to work in Office 2013. Try it in Office 2016. I cannot create issues for every application I am working in. This is not how it should be dealt with issues. This one is the same in more applications and I am sure that you can reproduce it in office 2016, Firefox 56 and later, Internet explorer, Edge and so on.

**Comment by fernando-jose-silva** [comment]
I did some tests:
open the word 2016.
now press nvda + page up or page down until exproration of screen.
In the laptop layout press nvda + up arrow and nvda + down arrow.
expected is that the nvda read the window.
The thing is that the nvda reads blank.

Another example:
open the start menu.
press page up and page down until exproration of screen.
In  nvda + arrows up or down.
expected is for nvda to read the start menu items as they appear on the screen.
what happens is that the nvda reads only in blank.

I think this next example is expected due to the limitations of nvda, but it would be very useful if it was made available.
Open the news app.
Press page up or page down until exproration of screen..
Press nvda + arrow up or down.
expected is that nvda read the objects as they appear in the window.
What happens is that the nvda reads only in blank

nvda next 14617
windows 10 build 16299.19 64bits

**Comment by derekriemer** [comment]
Try this in explorer. Modern apps just don't use display model, so screen review isn't as useful. @feerrenrut I've been thinking for a while of making screen mode use prev/next in flow order like touch screens for apps where it is blank, thoughts?

**Comment by fernando-jose-silva** [comment]
Screen navigation does not work in office 2016 which is not a modern application.
If any of the programming has missed me, I'm sorry.
In modern applications, nvda reads only blank when trying to navigate the screen.

**Comment by Adriani90** [comment]
closing this in favor of #3539 which actually covers this problem. Since this seems to depend on display model, I guess solving that issue will improve screen review in general.
